receiver/splunkhecreceiver/receiver.go

```diff
@@ -480,3 +492,29 @@ func isFlatJSONField(field interface{}) bool {
 	}
 	return true
 }
+func appendMetadataToResource(resource pcommon.Resource, req *http.Request, hecAttr splunk.HecToOtelAttrs) {
+	for k, v := range metadataMap {
+		if q := req.URL.Query().Get(k); q != "" {
+			key := getMetadataKey(hecAttr, k, v)
+			resource.Attributes().PutStr(key, q)
+		}
+	}
```

Comment:
- Can we not use a map and manually picks query params? Using a map in data path has minor performance penalty. If there are only 3 params, it won't look too bad to manually pick them out. 
- Today the logic to put these attributes into resource is in `splunkHecToLogData`. If we do it separately here there will be two places putting these metadata into pLog. I would recommend refactoring them so the logic is at the same place. It is just safer in case future maintainer changes one place but not the other.